spec/save_operation_callbacks_spec.cr

```diff
@@ -1,35 +1,21 @@
 require "./spec_helper"
 
 private class CallbacksSaveOperation < Post::SaveOperation
+  needs rollback : Bool = false
+
   @callbacks_that_ran = [] of String
   getter callbacks_that_ran
 
-  before_save :run_before_save
-  before_save run_before_save_again
-
-  before_create :run_before_create
-  before_create run_before_create_again
+  before_save setup_required_attributes
 
-  before_update :run_before_update
-  before_update :run_before_update_again
+  before_save :run_before_save
+  before_save { run_before_save_again }
 
   after_save :run_after_save
   after_save run_after_save_again
 
-  after_create :run_after_create
-  after_create run_after_create_again
-
-  after_update :run_after_update
-  after_update :run_after_update_again
```

Comment:
Removing all the `_update|_create` callbacks. Instead use different objects. If  we need something like this later we can add methods to `SaveOperation` like `creating?` and `updating?` so you can use them inside the callbacks if really necessary